src/psyclone/tests/psyir/tools/dependency_tools_test.py

```diff
@@ -172,25 +175,115 @@ def test_arrays_parallelise(parser):
     assert "Variable 'mask' is written to, and does not depend on the loop "\
            "variable 'jj'" in dep_tools.get_all_messages()[0]
 
+    jj_symbol = loops.scope.symbol_table.lookup("jj")
     # Write to array that does not depend on parallel loop variable
-    parallel = dep_tools.can_loop_be_parallelised(loops[1], "jj")
+    parallel = dep_tools.can_loop_be_parallelised(loops[1], jj_symbol)
     assert parallel
     assert not dep_tools.get_all_messages()
 
     # Use parallel loop variable in more than one dimension
-    parallel = dep_tools.can_loop_be_parallelised(loops[2], "jj")
+    parallel = dep_tools.can_loop_be_parallelised(loops[2], jj_symbol)
     assert not parallel
-    assert "Variable 'mask' is using loop variable 'jj' in index '(0, 0)'' " \
-        "and '(0, 1)'" in dep_tools.get_all_messages()[0]
+    assert "Variable 'mask' is written to and the loop variable 'jj' is " \
+           "used differently: mask(jj,jj) and mask(jj,jj)" \
+           in dep_tools.get_all_messages()[0]
 
     # Use a stencil access (with write), which prevents parallelisation
-    parallel = dep_tools.can_loop_be_parallelised(loops[3], "jj")
+    parallel = dep_tools.can_loop_be_parallelised(loops[3], jj_symbol)
     assert not parallel
     assert "Variable mask is written and is accessed using indices jj + 1 "\
            "and jj and can therefore not be parallelised" \
            in dep_tools.get_all_messages()[0]
 
 
+# -----------------------------------------------------------------------------
+def test_array_access_consistent(parser):
+    '''Tests the array checks of can_loop_be_parallelised.
+    '''
+    # pylint: disable=too-many-locals
+    reader = FortranStringReader('''program test
+                                 integer ji, jj, jk
+                                 integer, parameter :: jpi=5, jpj=10
+                                 real, dimension(jpi,jpi) :: a, b, c
+                                 do jj = 1, jpj   ! loop 0
+                                    do ji = 1, jpi
+                                       a(ji, jj) = -1.0d0
+                                     end do
+                                 end do
+                                 do jj = 1, jpj   ! loop 1
+                                    do ji = 1, jpi
+                                       a(ji, jj) = b(ji, jj)
+                                       c(ji, jj) = a(ji, jj)
+                                     end do
+                                 end do
+                                 do jj = 1, jpj   ! loop 2
+                                    do ji = 1, jpi
+                                       a(jj, ji) = b(ji, jj)
+                                     end do
+                                 end do
+                                 end program test''')
+    prog = parser(reader)
```

Comment:
I think you can use the `fortran_reader` fixture here. Just add it as an argument to the test and then you can do:

    psyir = fortran_reader('''program ... ...end program''')

Reply:
I can't: if I do this, I don't get loops created by Nemo API, and as a result, these loops do not have a type (latitude, ... etc) - and will therefore be rejected by the dependency tools. I could of course disable the loop type test, but I think it's useful to have that.

Reply:
Oh yes, I forgot about that, sorry. (You could raise the PSyIR provided by the fortran reader but I'm happy for you to leave it as it is.)